examples/plot_adversarial_mitigation.py

```diff
@@ -0,0 +1,367 @@
+import torch
+from math import sqrt
+
+# %%
+# Example 1: Simple use case with UCI Adult Dataset
+# =================================================
+# Firstly, we cover the most basic application of adversarial mitigation.
+# We start by loading and preprocessing the dataset.
+#
+# For this example we choose the feature 'sex' as the sensitive feature.
+# %%
+# Fetch the data
+X, y = fetch_openml(data_id=1590, as_frame=True, return_X_y=True)
+pos_label = y[0]
+
+# Choose sensitive feature
+z = X["sex"]
+
+# %%
+# The UCI adult dataset can not be fed into a neural network (yet),
+# as we have many columns that are not numerical in nature. To resolve this
+# issue, we could for instance use one-hot-encodings to preprocess categorical
+# columns. Additionally, let's preprocess the columns of number to a
+# standardized range. For these tasks, we can use functionality from
+# `sklearn.preprocessor`. Also, if we don't want to blatantly transform
+# NaN's (which are common in this dataset) to zero's, then we should take
+# care and define our own missing data imputer.
+
+
+ct = make_column_transformer(
+    (
+        Pipeline(
+            [
+                ("imputer", SimpleImputer(strategy="mean")),
+                ("normalizer", StandardScaler()),
+            ]
+        ),
+        make_column_selector(dtype_include=number),
+    ),
+    (
+        Pipeline(
+            [
+                ("imputer", SimpleImputer(strategy="most_frequent")),
+                ("encoder", OneHotEncoder(drop="if_binary", sparse=False)),
+            ]
+        ),
+        make_column_selector(dtype_include="category"),
+    ),
+)
+
+X_train, X_test, Y_train, Y_test, Z_train, Z_test = train_test_split(
+    X, y, z, test_size=0.2, random_state=12345, stratify=y
+)
+
+X_prep_train = ct.fit_transform(X_train)
+X_prep_test = ct.transform(X_test)
+
+
+# %%
+# Now, we can use :class:`fairlearn.adversarial.AdversarialFairnessClassifier`
+# to train on the
+# UCI Adult dataset. As our predictor and adversary models, we use for
+# simplicity the default constructors for fully connected neural
+# networks with sigmoid activations implemented in Fairlearn. We initialize
+# neural network constructors
+# by passing a list :math:`h_1, h_2, \dots` that indicate the number of nodes
+# :math:`h_i` per hidden layer :math:`i`. You can also put strings in this list
+# to indicate certain activation functions, or just pass an initialized
+# activation function directly.
+#
+# The specific fairness
+# objective that we choose for this example is demographic parity, so we also
+# set :code:`objective = "demographic_parity"`. We generally follow sklearn API,
+# but in this case we require some extra kwargs. In particular, we should
+# specify the number of epochs, batch size, whether to shuffle the rows of data
+# after every epoch, and optionally after how many seconds to show a progress
+# update.
+
+
+mitigator = AdversarialFairnessClassifier(
+    backend="tensorflow",
```

Comment:
I thought we wanna do a pytorch example.

Reply:
yea we did say this indeed